tf2onnx/schemas.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT license.
+
+"""
+tf2onnx.schema
+"""
+
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from collections import defaultdict, OrderedDict
+from onnx import defs
+
+
+ONNX_DOMAIN = ""
+
+
+class ONNXOpSchema(object):
+    """Wrapper for ONNX schema."""
+
+    def __init__(self, name, domain, since_version, attributes):
+        """Create a ONNX schema
+        Args:
+            name (str): op name
+            attributes (List[str]): valid attributes
+            domain (str): default value "" means it's ONNX domain
+            since_version (int): opset version, default is 1
+        """
+        self._name = name
+        self._domain = domain
+        self._attributes = attributes
+        self._since_version = since_version
+
+    @property
+    def attributes(self):
+        return self._attributes
+
+
+def map_schemas(onnx_schemas):
+    """Map schema in ONNX to tf2onnx.ONNXOpSchema"""
+    name_domain_version_schema_map = defaultdict(lambda: defaultdict(dict))
+    for s in onnx_schemas:
+        name = s.name
+        domain = s.domain
+        since_version = int(s.since_version)
+        name_domain_version_schema_map[name][domain][since_version] = \
+            ONNXOpSchema(name, domain, since_version, s.attributes)
+
+    ordered_version_map = defaultdict(lambda: defaultdict(OrderedDict))
+    for name, domain_version_schema_map in name_domain_version_schema_map.items():
+        for domain, version_schema_map in domain_version_schema_map.items():
+            ordered_version_map[name][domain] = OrderedDict(
+                sorted(
+                    version_schema_map.items(),
+                    key=lambda x: x[0]
+                )
+            )
+    return ordered_version_map
+
+
+all_schemas = defs.get_all_schemas_with_history()
+# format is <OpName, <Domain, <SinceVersion, OpSchema>>>
+schemas = map_schemas(all_schemas)
+
+
+def get_schema(name, max_inclusive_opset_version, domain=ONNX_DOMAIN):
+    """Get schema by name within specific version."""
+    if name not in schemas:
+        return None
+    if max_inclusive_opset_version < 1:
+        return None
+    domain_version_schema_map = schemas[name]
+    version_schema_map = domain_version_schema_map[domain]
+    versions = list(version_schema_map.keys())
+    for i, v in enumerate(versions):
+        if max_inclusive_opset_version < v:
```

Comment:
you may want to store version from high to low, so if version <= max_inclusive_opset_version, return.
can make the code here simpler.